### PR TITLE
Agent networking

### PR DESCRIPTION
**Pull request by agovind92**
Add agent network peer and connectivity to connect via hub bastion 

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ x] No
```

### CODE REVIEW

File: terraform/vars/adoagent.tfvars [hunk 1]
```diff
@@ -22,4 +22,18 @@ nsg_name = "vh-infra-core-ado-nsg"
 # environment
 env = "stg"
 
-dns_zone = ["dev.platform.hmcts.net", "test.platform.hmcts.net", "ithc.platform.hmcts.net", "staging.platform.hmcts.net", "platform.hmcts.net", "demo.platform.hmcts.net"]
\ No newline at end of file
+dns_zone = ["dev.platform.hmcts.net", "test.platform.hmcts.net", "ithc.platform.hmcts.net", "staging.platform.hmcts.net", "platform.hmcts.net", "demo.platform.hmcts.net"]
```
Comment (chris-pateman) [inline terraform/vars/adoagent.tfvars:25]: missing the `sandbox` DNS and also the `hearings.reform.hmcts.net`

File: terraform/virtual-network.tf [hunk 2]
```diff
@@ -58,6 +65,14 @@ resource "azurerm_virtual_network_peering" "vh_infra_core_ado_TO_hubs" {
   provider                  = azurerm.current_sub_peer
 }
 
+resource "azurerm_virtual_network_peering" "core-infra-vnet-mgmt_TO_vh_infra_core_ado" {
+  name                      = "core-infra-vnet-mgmt-TO-vh-infra-core-ado"
```
Comment (chris-pateman) [inline terraform/virtual-network.tf:69]: can these be variables e.g. "${var.core_infra_vnet_name}-TO-${var.ado_vnet_name}"

**Review by chris-pateman — CHANGES_REQUESTED** [review]

File: terraform/key-vault.tf [hunk 3]
```diff
@@ -36,4 +36,17 @@ resource "azurerm_key_vault" "vh_infra_core_ado" {
     ]
   }
   tags = local.common_tags
+}
+resource "azurerm_key_vault_access_policy" "temp" {
+  key_vault_id = azurerm_key_vault.vh_infra_core_ado.id
+  tenant_id    = data.azurerm_client_config.current.tenant_id
+  object_id    = "216a0f7e-97b3-4ebc-ad19-9ccdd0f59077"
```
[the remark this answers is not in the document]
Comment (agovind92) [reply terraform/key-vault.tf:43]: this has been removed, was only needed temp so i could read the p/w

File: terraform/vars/adoagent.tfvars (hunk 1 above)
Comment (agovind92) [reply terraform/vars/adoagent.tfvars]: these do not exist

**Review by chris-pateman — CHANGES_REQUESTED** [review]

**Pull request merged**